### ISSUE TITLE
Nightlights Data Extraction in R

### ISSUE THREAD

**Issue by chrisvwn** [ropensci/onboarding#159]
### Summary

-   What does this package do? (explain in 50 words or less):
The Rnightlights package extracts raster and zonal statistics from satellite nightlight rasters downloaded from the United States National Oceanic and Atmospheric Administration (<http://www.noaa.gov>) free data repositories. Statistics are calculated at the country and sub-country levels.

-   Paste the full DESCRIPTION file inside a code block below:

```
Package: Rnightlights
Version: 0.1.3
Date: 2017-11-21
Title: Satellite Nightlight Data Extraction
Description: The Rnightlights package extracts raster and zonal statistics
    from satellite nightlight rasters downloaded from the United States 
    National Oceanic and Atmospheric Administration (<http://www.noaa.gov>) 
    free data repositories. Both the DMSP-OLS annual and SNPP-VIIRS monthly 
    nightlight raster data are supported. Satellite nightlight raster tiles are 
    downloaded and cropped to the country boundaries using shapefiles from the GADM
    database of Global Administrative Areas (<http://gadm.org>). Zonal statistics
    are then calculated at the lowest administrative boundary for the selected
    country and cached locally for future retrieval. Finally, a simple data
    explorer/browser is included that allows one to visualize the cached data e.g.
    graphing, mapping and clustering regional data.
Authors@R: c(person(given="Christopher", family="Njuguna", role=c("aut", "cre", "cph"), email="chris.njuguna@gmail.com"),
            person(given="Henrik", family="Bengtsson", role="ctb", email="henrikb@braju.com", comment="data setup, https://cran.r-project.org/package=R.cache"),
            person(given="Black", family="Guru", role="ctb", comment="gdal zonal, http://www.guru-gis.net/efficient-zonal-statistics-using-r-and-gdal/"),
            person(given="Jeff", family="Chen", role="ctb", comment="masq function, https://commercedataservice.github.io/tutorial_viirs_part1/"),
            person(given="Star", family="Ying", role="ctb", comment="masq function, https://commercedataservice.github.io/tutorial_viirs_part1/"),
            person(given="Chris", family="Elvidge", role="ctb", comment="information + masq function"))
License: GPL-3
Imports:
    cleangeo,
    compiler,
    data.table,
    doParallel,
    dplyr,
    foreach,
    gdalUtils,
    lubridate,
    methods,
    raster,
    readr,
    reshape2,
    rgdal,
    rgeos,
    R.utils,
    rvest,
    rworldmap,
    settings,
    sp,
    stats,
    stringr,
    utils,
    xml2
Suggests:
    dendextend,
    ggdendro,
    ggplot2,
    leaflet,
    plotly,
    RColorBrewer,
    reshape,
    shiny,
    shinydashboard
SystemRequirements:
    gdal
    wget
    curl
    aria2
RoxygenNote: 6.0.1
BugReports: https://github.com/chrisvwn/Rnightlights/issues
URL: https://github.com/chrisvwn/Rnightlights

```

-   URL for the package (the development repository, not a stylized html page):
http://github.com/chrisvwn/Rnightlights

- Please indicate which category or categories from our [package fit policies](https://github.com/ropensci/onboarding/blob/master/policies.md#package-fit) this package falls under ***and why**(? (e.g., data retrieval, reproducibility. If you are unsure, we suggest you make a pre-submission inquiry.):
[e.g., "data extraction, because the package parses a scientific data file format"]
data retrieval, because the package downloads the appropriate satellite nightlight raster tiles given time period and country ISO3 code
data extraction, because the package clips satellite nightlight imagery to country boundaries and performs aggregation functions on the rasters
geospatial data, because the package deals with satellite nightlight geospatial raster tiles

-   Who is the target audience?  
Researchers and anyone who would like to perform analyses of satellite nightlights imagery.

-   Are there other R packages that accomplish the same thing? If so, how does
yours differ or meet [our criteria for best-in-category](https://github.com/ropensci/onboarding/blob/master/policies.md#overlap)?
There are no R packages that accomplish the same thing that I am aware of.

### Requirements

Confirm each of the following by checking the box.  This package:

- [x] does not violate the Terms of Service of any service it interacts with. 
- [x] has a CRAN and OSI accepted license.
- [x] contains a README with instructions for installing the development version. 
- [x] includes documentation with examples for all functions.
- [x] contains a vignette with examples of its essential functions and uses.
- [ ] has a test suite.
- [x] has continuous integration, including reporting of test coverage, using services such as Travis CI, Coeveralls and/or CodeCov.
- [x] I agree to abide by [ROpenSci's Code of Conduct](https://github.com/ropensci/onboarding/blob/master/policies.md#code-of-conduct) during the review process and in maintaining my package should it be accepted.

#### Publication options

- [x] Do you intend for this package to go on CRAN?  
- [x] Do you wish to automatically submit to the [Journal of Open Source Software](http://joss.theoj.org/)? If so:
    - [x] The package has an **obvious research application** according to [JOSS's definition](http://joss.theoj.org/about#submission_requirements).
    - [x] The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with a high-level description in the package root or in `inst/`.
    - [ ] The package is deposited in a long-term repository with the DOI: 
    - (*Do not submit your package separately to JOSS*)

### Detail

- [x] Does `R CMD check` (or `devtools::check()`) succeed?  Paste and describe any errors or warnings:

- [x] Does the package conform to [rOpenSci packaging guidelines](https://github.com/ropensci/onboarding/blob/master/packaging_guide.md)? Please describe any exceptions:

- If this is a resubmission following rejection, please explain the change in circumstances:

- If possible, please provide recommendations of reviewers - those with experience with similar packages and/or likely users of your package - and their GitHub user names:

**Comment by sckott** [comment]
Thanks for your submission @chrisvwn ! Editors are discussing now - we'll get back to you asap.

**Comment by sckott** [comment]
@chrisvwn Submission is a good fit for rOpenSci. We'll assign an editor asap

**Comment by chrisvwn** [comment]
@sckott Thank you!!

**Comment by maelle** [comment]
@chrisvwn before starting the review process your package must:

* have a test suite 😉  Do you need help with this? I can recommend reading http://r-pkgs.had.co.nz/tests.html and then ask me any question, for that open an issue in your repo (to keep this thread clean) or create a thread in [our forum](https://discuss.ropensci.org/). Don't worry, we'll get you there!

* have a low case name as advised in our guidelines (since your package isn't on CRAN yet and I imagine doesn't have a large userbase yet?): it's easier for users to remember e.g. "rnightlights" and for the sake of simplicity you could even call it "nightlights". I changed the name of my own "Ropenaq" package to "ropenaq" to comply and I can say it wasn't very difficult.

I'll put this submission on hold because of this.

I also have 3 remarks/questions:

* Any reason why you use `reshape` and not `tidyr` or `reshape2`?

* Add README.md, the Travis config file etc to .Rbuildignore 

* I'd rephrase
```
Enter 0 to use a temporary directory for this session only. 

1: Create data path under home directory 'C:/Users/Maelle/Documents
2: Choose a different directory as the data path
```

to 

```
0: Use a temporary directory for this session only. 
1: Create data path under home directory 'C:/Users/Maelle/Documents
2: Choose a different directory as the data path
```

**Comment by maelle** [comment]
Oh sorry,  if it is on CRAN don't change the name ☺

**Comment by maelle** [comment]
Closing this as discussed in the package repo! When submitting your package again after it's stable please put a link to this issue. See you soon in this repository! 😉
